router/core/src/main/scala/io/buoyant/router/DynParamsFactory.scala

```diff
@@ -0,0 +1,125 @@
+package io.buoyant.router
+
+import com.twitter.finagle
+import com.twitter.finagle.Stack.{Leaf, Node}
+import com.twitter.finagle._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.util._
+
+object HaltClosePropagationFactory {
+  def role: Stack.Role = Stack.Role("HaltClose")
+
+  case class Param(closer: Promise[Time])
+  implicit object Param extends Stack.Param[Param] {
+    override def default: Param = Param(new Promise[Time])
+  }
+
+  def module[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] = new finagle.Stack.Module1[Param, ServiceFactory[Req, Rep]] {
+    override def role: Stack.Role = HaltClosePropagationFactory.role
+    override def description: String = "Stop service factory closes from propagating down the stack"
+    override def make(
+      params: Param,
+      next: ServiceFactory[Req, Rep]
+    ): ServiceFactory[Req, Rep] =
+      new HaltClosePropagationFactory[Req, Rep](next, params.closer)
+  }
+}
+
+class HaltClosePropagationFactory[Req, Rep](underlying: ServiceFactory[Req, Rep], closer: Promise[Time]) extends ServiceFactory[Req, Rep] {
```

Comment:
TIOLI: I think we might save a tiny bit of boilerplate by extending ServiceFactoryProxy